### PR TITLE
Add foreign configuration support

### PR DESCRIPTION
**Pull request by dgilbert101**
The following pull request adds methods to deal with foreign configurations:
- has_foreign_configurations
- import_foreign_configurations
- delete_foreign_configurations

### CODE REVIEW

File: pystorcli2/foreignconfig/__init__.py [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+'''
+
+from .. import StorCLI
+from .. import common
+from .. import controller
+from .. import exc
+
+from typing import Optional
+
+class ForeignConfigurations(object):
```
[the remark this answers is not in the document]
Comment (ralequi) [reply pystorcli2/foreignconfig/__init__.py:17]: Hi,
Sure, totally agree with @ulmitov .
I don't clearly see the benefit of an independent object when it only provides 3 useful methods (has, import & delete FC)

[the remark this answers is not in the document]
Comment (dgilbert101) [reply pystorcli2/foreignconfig/__init__.py:17]: ok, i can refactor. when I originally started, I expected the /fx functionality to work. Only to find out it is just /fall.

[the remark this answers is not in the document]
Comment (dgilbert101) [reply pystorcli2/foreignconfig/__init__.py]: The docs are incorrect. If you look at the output from the latest versions of storcli or perccli, you can only use /fall

    $ /opt/MegaRAID/perccli/perccli64 /fall help
    PercCli SAS Customization Utility Ver 007.1623.0000.0000 May 17, 2021
    (c)Copyright 2021, Broadcom Inc. All Rights Reserved.

    perccli /cx/fall show [all] [securityKey = xxx]
    perccli /cx/fall del|delete [securityKey = xxx]
    perccli /cx/fall import [preview] [securityKey = xxx]

    $ /opt/MegaRAID/storcli/storcli64 /fall help
    StorCli SAS Customization Utility Ver 007.2408.0000.0000 Nov 15, 2022
    (c)Copyright 2022, Broadcom Inc. All Rights Reserved.

    storcli /cx/fall show [all] [securityKey = xxx]
    storcli /cx/fall del|delete [securityKey = xxx]
    storcli /cx/fall import [preview] [securityKey = xxxx

File: pystorcli2/controller/__init__.py [hunk 2]
```diff
@@ -416,6 +419,53 @@ def set_cc(self, value, starttime=None):
...
+
+    def import_foreign_configurations(self, securitykey: Optional[str] = None):
+        """Imports foreign configurations
+
+        Returns:
+            (dict): response cmd data
+        """
+        args = [
+            '/fall',
+            'del'
```
Comment (ralequi) [inline pystorcli2/controller/__init__.py:462]: maybe you want to say `import` ?

Comment (dgilbert101) [reply pystorcli2/controller/__init__.py]: hah, yeah, definitely should

File: pystorcli2/controller/__init__.py [hunk 3]
```diff
@@ -416,6 +419,53 @@ def set_cc(self, value, starttime=None):
 
         return common.response_setter(self._run(args))
 
+    @property
+    def has_foreign_configurations(self) -> bool:
+        """(bool): true if controller has foreign configurations
+        """
+        args = [
+            '/fall',
+            'show'
```
[the remark this answers is not in the document]
Comment (dgilbert101) [reply pystorcli2/controller/__init__.py:427]: I've added it, but I currently have no way to test this.

**Review by ralequi — APPROVED** [review]

**Review by ralequi — CHANGES_REQUESTED** [review]
It seems that some /fall access causes raising exceptions.

We would have to handle those things. Check issue #13

**Review by ralequi — APPROVED** [review]
I think everything is OK now. @ulmitov something to add?

**Comment by ralequi** [comment]
Thx, Let me check

**Comment by ralequi** [comment]
Ok, I'm going to merge this and finish it on develop branch

Thanks @dgilbert101 & @ulmitov

**Pull request merged**